packages/service-client/odsp-client/src/OdspClient.ts

```diff
@@ -0,0 +1,201 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IDocumentServiceFactory } from "@fluidframework/driver-definitions";
+import {
+	OdspDocumentServiceFactory,
+	OdspDriverUrlResolver,
+	createOdspCreateContainerRequest,
+} from "@fluidframework/odsp-driver";
+import {
+	type ContainerSchema,
+	DOProviderContainerRuntimeFactory,
+	IFluidContainer,
+	FluidContainer,
+	IRootDataObject,
+} from "@fluidframework/fluid-static";
+import {
+	AttachState,
+	IContainer,
+	IFluidModuleWithDetails,
+} from "@fluidframework/container-definitions";
+import { IClient } from "@fluidframework/protocol-definitions";
+import { Loader } from "@fluidframework/container-loader";
+import { v4 as uuid } from "uuid";
+import {
+	IOdspResolvedUrl,
+	OdspResourceTokenFetchOptions,
+} from "@fluidframework/odsp-driver-definitions";
+import { ITokenResponse } from "@fluidframework/azure-client";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { IRequest } from "@fluidframework/core-interfaces";
+import { assert } from "@fluidframework/core-utils";
+import {
+	OdspClientProps,
+	OdspConnectionConfig,
+	OdspContainerServices,
+	OdspServiceAttributes,
+} from "./interfaces";
+import { OdspAudience } from "./OdspAudience";
+
+/**
+ * @alpha
```

Comment:
Do we expect external users to subclass this at all? If not, we should annotate this as `@sealed`.